my_2d_robot/src/calibration_2d.cpp

```diff
@@ -0,0 +1,130 @@
+#include "ceres/ceres.h"
+#include "glog/logging.h"
+#include "yaml-cpp/yaml.h"
+#include "ros/ros.h"
+#include "std_msgs/String.h"
+#include <sensor_msgs/JointState.h>
+#include <vector>
+
+using ceres::AutoDiffCostFunction;
+using ceres::CostFunction;
+using ceres::Problem;
+using ceres::Solve;
+using ceres::Solver;
+
+struct OffsetCalibration
+{
+public:
+  OffsetCalibration(double offset_joint1, double offset_joint2, double true_x, double true_y)
+    : true_x_(true_x), true_y_(true_y)
+  {
+    // Link lengths of robot (specified in launch file).
+    ros::NodeHandle n;
+    n.getParam("link_1", link1_);
```

Comment:
Lets get this elsewhere and pass it into the struct constructor as a paramater. It would be cleanest to get all params in main.

Reply:
Is it alright to have another constructor for this? Since I'm not too sure how I would pass the other parameters into the current constructor....unless I pass it to the calibrate function as a parameter as well and then pass it to the constructor in the add residual block?